Library/brew.rb

```diff
@@ -39,13 +38,24 @@
   EOS
 end
 
+# Check for user agreement of the Xcode license before permitting
+# any other brew usage to continue. This prevents the situation where
+# people are instructed to "please re-run as root via sudo" on brew commands.
+# The check can only fail when Xcode is installed & the active developer dir.
+if `/usr/bin/xcrun clang 2>&1` =~ /license/ && !$?.success?
+  odie <<-EOS.undent
+    You have not agreed to the Xcode license. Please resolve this by running:
+      sudo xcodebuild -license
+  EOS
+end
```

Comment:
@xu-cheng If you want to play with a few ideas I don't mind if you steal the PR. I'll add the Xcode check in here and we can either move ahead with this PR or whatever ideas you've in mind after the CI has trundled back over this.